### DISCUSSION TITLE
Increase in slow pages since switching to app directory

### DISCUSSION THREAD

**Discussion by JamesSingleton** [vercel/next.js#56754] [Help]
### Summary

I switched to the app directory September 5th and since then, ahrefs has reported an increasing amount of slow pages. I deploy to Vercel, my CMS is Sanity and before switching to the app directory I didn’t have any slow pages or barely any. I will say my paginated pages aren’t statically generated due to using `?page=n` for pagination. However, the blog posts are generated via `generateStaticParams` but are still showing up as slow. Any ideas?

![image](https://github.com/vercel/next.js/assets/21000200/407d5794-51d9-435c-8751-f41d9f414792)

**Comment by icyJoseph** [comment]
Hi,

I have but one thing in mind at this point, that maybe the RSC+JS+HTML+CSS bundle is far too large? I remember seeing a video where the wrong props passing strategy really sunk the performance: 

- https://www.youtube.com/watch?v=FfHsIio4aCU

Perhaps try the tool he shows? See if the payload has become far too large.

**Reply by JamesSingleton** [reply]
Interesting 🤔 so I ran `oha` against localhost:3000 first on one of the URLs it complained about and got

```
Summary:
  Success rate:	100.00%
  Total:	10.0012 secs
  Slowest:	0.1775 secs
  Fastest:	0.0451 secs
  Average:	0.0617 secs
  Requests/sec:	807.1020

  Total data:	260.63 MiB
  Size/request:	33.06 KiB
  Size/sec:	26.06 MiB
```

Then I ran it against production (hosted on Vercel) and got

```
Summary:
  Success rate:	100.00%
  Total:	10.0007 secs
  Slowest:	3.6348 secs
  Fastest:	0.0623 secs
  Average:	0.4542 secs
  Requests/sec:	106.8921

  Total data:	27.96 MiB
  Size/request:	26.79 KiB
  Size/sec:	2.80 MiB
```

Also using his tester that he created I get this:

![Screenshot 2023-10-17 at 10 02 36 PM](https://github.com/vercel/next.js/assets/21000200/5a3d5f01-aa50-4b7f-92c5-d8a34b4f39d7)

So it looks like it _might_ be vercel?

**Reply by icyJoseph** [reply]
Right, there's very little data being shipped, so I wouldn't expect the speed complains to be related to page painting, but rather response times, backed up by the oha tests.

Does this mean that waking up the lambda, or serving the initial requests is under performing? 🤔

**Reply by JamesSingleton** [reply]
Maybe a little bit of both? I ran oha against production 3 times and the slowest was still in like 2.5 seconds. These tests were back to back so I would assume the lambda wasn’t asleep.
